circus-train-comparator/src/main/java/com/hotels/bdp/circustrain/comparator/hive/wrappers/PathMetadata.java

```diff
@@ -71,7 +72,11 @@ public int getChecksumLength() {
   }
 
   public byte[] getChecksum() {
-    return checksum;
+    if (checksum == null) {
+      return new byte[] {};
```

Comment:
The old behaviour would have returned `null` here so we need to be careful that all calling code can deal with this change (or return null again).

Reply:
You make a good point. I'll change it to return `null`.